x/superfluid/keeper/msg_server_test.go

```diff
@@ -192,3 +193,191 @@ func (suite *KeeperTestSuite) TestMsgLockAndSuperfluidDelegate() {
+				continue
+			} else {
+				suite.Require().NoError(err)
+				assertEventEmitted(suite, suite.Ctx, types.TypeEvtSuperfluidUndelegate, 1)
+			}
+		}
+	}
+}
+
+// TestMsgSuperfluidUnbondLock_Event tests that events are correctly emitted
+// when calling SuperfluidUnbondLock.
+func (suite *KeeperTestSuite) TestMsgSuperfluidUnbondLock_Event() {
+	suite.SetupTest()
+	msgServer := keeper.NewMsgServerImpl(suite.App.SuperfluidKeeper)
+
+	// setup validators
+	valAddrs := suite.SetupValidators([]stakingtypes.BondStatus{stakingtypes.Bonded})
+
+	denoms, _ := suite.SetupGammPoolsAndSuperfluidAssets([]sdk.Dec{sdk.NewDec(20), sdk.NewDec(20)})
+
+	// setup superfluid delegations
+	_, _, locks := suite.setupSuperfluidDelegations(valAddrs, []superfluidDelegation{{0, 0, 0, 1000000}}, denoms)
+
+	for _, lock := range locks {
+		startTime := time.Now()
+		sender, _ := sdk.AccAddressFromBech32(lock.Owner)
+
+		// first we test that SuperfluidUnbondLock would cause error before undelegating
+		_, err := msgServer.SuperfluidUnbondLock(sdk.WrapSDKContext(suite.Ctx), types.NewMsgSuperfluidUnbondLock(sender, lock.ID))
+		suite.Require().Error(err)
+
+		// undelegation needs to happen prior to SuperfluidUnbondLock
+		err = suite.App.SuperfluidKeeper.SuperfluidUndelegate(suite.Ctx, lock.Owner, lock.ID)
+		suite.Require().NoError(err)
+
+		// test SuperfluidUnbondLock
+		unbondLockStartTime := startTime.Add(time.Hour)
+		suite.Ctx = suite.Ctx.WithBlockTime(unbondLockStartTime)
+		_, err = msgServer.SuperfluidUnbondLock(sdk.WrapSDKContext(suite.Ctx), types.NewMsgSuperfluidUnbondLock(sender, lock.ID))
+		suite.Require().NoError(err)
+		assertEventEmitted(suite, suite.Ctx, types.TypeEvtSuperfluidUnbondLock, 1)
+	}
+}
+
+// TestMsgUnPoolWhitelistedPool_Event tests that events are correctly emitted
+// when calling UnPoolWhitelistedPool.
+func (suite *KeeperTestSuite) TestMsgUnPoolWhitelistedPool_Event() {
+	suite.SetupTest()
+	msgServer := keeper.NewMsgServerImpl(suite.App.SuperfluidKeeper)
+
+	// setup validators
+	valAddrs := suite.SetupValidators([]stakingtypes.BondStatus{stakingtypes.Bonded})
+
+	denoms, poolIds := suite.SetupGammPoolsAndSuperfluidAssets([]sdk.Dec{sdk.NewDec(20)})
+
+	// whitelist designated pools
+	suite.App.SuperfluidKeeper.SetUnpoolAllowedPools(suite.Ctx, poolIds)
+
+	// setup superfluid delegations
+	_, _, locks := suite.setupSuperfluidDelegations(valAddrs, []superfluidDelegation{{0, 0, 0, 1000000}}, denoms)
+
+	for index, poolId := range poolIds {
+		sender, _ := sdk.AccAddressFromBech32(locks[index].Owner)
+		suite.Ctx = suite.Ctx.WithBlockHeight(v8constants.UpgradeHeight)
+		_, err := msgServer.UnPoolWhitelistedPool(sdk.WrapSDKContext(suite.Ctx), types.NewMsgUnPoolWhitelistedPool(sender, poolId))
+		suite.Require().NoError(err)
+		assertEventEmitted(suite, suite.Ctx, types.TypeEvtUnpoolId, 1)
+	}
+}
+
+func assertEventEmitted(suite *KeeperTestSuite, ctx sdk.Context, eventTypeExpected string, numEventsExpected int) {
```

Comment:
this function has been defined in `app/apptesting/events.go` pretty sure we can just use that

Reply:
Thanks for reminding